lib/wasix/src/syscalls/wasm.rs

```diff
@@ -26,6 +26,8 @@ pub fn platform_clock_time_get(
     clock_id: Snapshot0Clockid,
     precision: Timestamp,
 ) -> Result<i64, Errno> {
-    let new_time: DateTime<Local> = Local::now();
-    Ok(new_time.timestamp_nanos() as i64)
+    Local::now()
+        .timestamp_nanos_opt()
+        .map(|ts| ts as i64)
+        .ok_or(Errno::Overflow)
```

Comment:
I think it's the error that makes the most sense. 

This is what [the `chrono` docs](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.timestamp_nanos_opt) say:

> Returns the number of non-leap-nanoseconds since January 1, 1970 UTC.
> 
> ## [Panics](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#panics-4)
> An i64 with nanosecond precision can span a range of ~584 years. This function panics on an out of range DateTime.
> 
> The dates that can be represented as nanoseconds are between 1677-09-21T00:12:44.0 and 2262-04-11T23:47:16.854775804.